libevhtp: Fix dependencies not selected when building static library

jefferyto opened the pull request:
Maintainer: me (for seafile-server), @commodo, @cotequeiroz 
Compile tested: armvirt-64/x86-generic, 2020-04-28 snapshot sdk
Run tested: none

Description:
When libevhtp is added to a package's `PKG_BUILD_DEPENDS` (i.e. for seafile-server), libevhtp's target package dependencies are not automatically selected, in particular libevent2-openssl and
libevent2-pthreads.

Moreover, if libevent-openssl and libevent2-pthreads are not selected, OpenSSL and thread support are disabled when compiling libevent2, which in turn causes a compile error when building libevhtp.

This adds a config option, `LIBEVHTP_BUILD_DEPENDS`, that will select libevent2-openssl and libevent2-pthreads (via `LIBEVHTP_BUILD_SELECT`) when it is selected.

Other build dependencies are moved to `PKG_BUILD_DEPENDS`.

This also updates seafile-server to remove the (indirect) dependencies added in 13d843fec1cd3281cc469fc74de302c1f793b723 and instead select `LIBEVHTP_BUILD_DEPENDS`.

This also updates the maintainer's email address and adds myself as another maintainer.

Signed-off-by: Jeffery To <jeffery.to@gmail.com>

jefferyto commented:
@commodo Would you mind testing this to see if you still get the build error you got before?

commodo commented:
finished testing it now;
works fine

jefferyto commented:
Thanks @commodo :joy:

Apologies for forgetting to ping you @cotequeiroz :sweat_smile:

cotequeiroz commented on libs/libevhtp/Makefile:
```diff
@@ -44,6 +46,14 @@ define Package/libevhtp/description
+define Package/libevhtp/config
+config LIBEVHTP_STATIC_DEPENDS
+	tristate
+	default PACKAGE_libevhtp
```
```suggestion
	default PACKAGE_libevhtp

config LIBEVHTP_BUILD_DEPS
	tristate
	default m if LIBEVHTP_STATIC_DEPENDS
```

I suggest selecting the packages as `m`, so that libevent2-pthreads does not get included in the image if seafile-server (or even empty lbevhtp) is selected.

jefferyto replied:
I think with this change `LIBEVHTP_STATIC_DEPENDS` can be a `bool` and I'll see if I can think of some better names than "static depends" and "build depends", but this is a good idea.

jefferyto commented:
@commodo Would you want to be co-maintainer for this package? (It's okay to say no if you don't want to) :joy:

I'll replace @cotequeiroz with your name or update his email address; either way I'll add my name as well.

commodo commented:
> @commodo Would you want to be co-maintainer for this package? (It's okay to say no if you don't want to) joy
> 

answer is no :)
atm, i'm trying to reduce some of my work;
i'm way more caught-up in kernel work these days, and i can sometimes barely keep-up some packages-work;
also, i am still not a fan of seafile :)

jefferyto commented:
@commodo No problem, I completely understand :smile:

jefferyto commented:
Updated:
* Rename `LIBEVHTP_STATIC_DEPENDS` to `LIBEVHTP_BUILD_DEPENDS` (to match `PKG_BUILD_DEPENDS`) and changed to `bool`
* Add intermediate `LIBEVHTP_BUILD_SELECT` option to select libevent2-openssl and libevent2-pthreads (as `m`)
* Update @cotequeiroz's email address
* Add myself as additional maintainer
